--- apps/tools/vite/asset_loader_old.py	2025-05-02 13:17:05
+++ apps/tools/vite/asset_loader_new.py	2025-05-02 13:17:57
@@ -67,6 +67,9 @@
     css: Optional[List[str]] = []
     imports: Optional[List[str]] = []
     dynamicImports: Optional[List[str]] = []
+    # BEGIN: Added by TechSmart
+    assets: Optional[List[str]] = []
+    # END: Added by TechSmart
 
 
 class ManifestClient:
\ No newline at end of file
@@ -519,8 +522,22 @@
 
         manifest_entry = self.manifest.get(path)
 
-        return self.get_production_server_url(manifest_entry.file)
+        # BEGIN: Added by TechSmart
+        import os.path
+        (_, path_ext) = os.path.splitext(path)
+        if path_ext in ['.jpg', '.png', '.svg']:  # is image?
+            # Non JS asset
+            (manifest_file,) = manifest_entry.assets
+            assert manifest_file.endswith(path_ext)
+        else:
+            # JS asset
+            manifest_file = manifest_entry.file
+        # END: Added by TechSmart
 
+        # BEGIN: Changed by TechSmart
+        return self.get_production_server_url(manifest_file)
+        # END: Changed by TechSmart
+
     def generate_vite_legacy_polyfills(
         self,
         nomodule: bool = True,
\ No newline at end of file
